### PR TITLE
Fixed #131 - Validate actorName in make method

### LINKED ISSUE #131: Validate actorName in make method
The `make` method allows actor names that do not match `select`'s regex:

```scala
val regex = "^(?:zio:\\/\\/)(\\w+)[@](\\d+\\.\\d+\\.\\d+\\.\\d+)[:](\\d+)[/]([\\w|\\/]+)$".r
```

The same Regex should be used to validate the actor's full path before creating it.

----

Failing example:

```scala
for {
  system <- ActorSystem("actorSystem", Some("0.0.0.0", "9000"))
  actor  <- system.make("actor-123", Supervisor.none, Unit, stateful)
  _      <- system.select(s"zio://actorSytem@0.0.0.0:9000/actor-123") // fails with Exception: "No such actor in local ActorSystem"
} yield ()
```

### PR DESCRIPTION
**Pull request by gastonlucero**
Resolves #131

This PR improves actor name validation when an actor is created or selected.
The name is validated with :
 * regex that allows some special characters -_. * $ +: @ & =,! ~ ';
 * Not null name
 * Non Empty  name 
Also in the selection of nested actors (actorOne / childOne), the route is validated correctly.

### CODE REVIEW

File: src/main/scala/zio/actors/ActorSystem.scala [hunk 1]
```diff
@@ -222,8 +222,14 @@ final class ActorSystem private[actors] (
           obj      <- readFromWire(worker)
           envelope = obj.asInstanceOf[Envelope]
           actorMap <- refActorMap.get
-
-          _ <- actorMap.get("/" + envelope.recipient.split("/").last) match {
+          remoteActorPath <- for {
```
Comment (mijicd) [inline src/main/scala/zio/actors/ActorSystem.scala]: You can drop the entire for-comprehension with: `resolvePath(envelope.recipient).map(_._4)`

File: src/main/scala/zio/actors/ActorSystem.scala [hunk 2]
```diff
@@ -269,6 +279,17 @@ private[actors] object ActorSystemUtils {
         )
     }
 
+  def buildFinalName(parentActorName: String, actorName: String): Task[String] =
```
Comment (mijicd) [inline src/main/scala/zio/actors/ActorSystem.scala]: This should probably be private.

File: src/main/scala/zio/actors/ActorSystem.scala [hunk 3]
```diff
@@ -269,6 +279,17 @@ private[actors] object ActorSystemUtils {
...
+    actorName match {
+      case ""   => IO.fail(new Exception("Actor actor must not be empty"))
+      case null => IO.fail(new Exception("Actor actor must not be null"))
+      case name =>
+        regexName.pattern.matcher(name).matches match {
```
Comment (mijicd) [inline src/main/scala/zio/actors/ActorSystem.scala]: You can use regex directly:

```scala
actorName match {
  case null => ...
  case "" => ...
  case regexName(_*) => ..
  case _ => ...
}
```

File: src/main/scala/zio/actors/ActorSystem.scala [hunk 4]
```diff
@@ -251,10 +257,14 @@ final class ActorSystem private[actors] (
 /* INTERNAL API */
 
 private[actors] object ActorSystemUtils {
-  private val regex = "^(?:zio:\\/\\/)(\\w+)[@](\\d+\\.\\d+\\.\\d+\\.\\d+)[:](\\d+)[/]([\\w|\\/]+)$".r
+
+  private val regexName = "[\\w+|\\d+|(\\-_.*$+:@&=,!~';.)|\\/]+".r
```
Comment (mijicd) [inline src/main/scala/zio/actors/ActorSystem.scala]: Make constant names upper-camel case.

File: src/main/scala/zio/actors/ActorSystem.scala (hunk 3 above)
Comment (gastonlucero) [reply src/main/scala/zio/actors/ActorSystem.scala]: Great! I didn't know I could use it like that. Thanks

File: src/main/scala/zio/actors/ActorSystem.scala (hunk 2 above)
Comment (gastonlucero) [reply src/main/scala/zio/actors/ActorSystem.scala]: You mean private, but within the ActorSystem class?

**Pull request merged**